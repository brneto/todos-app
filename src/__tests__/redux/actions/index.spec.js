import {
  fetchTodos,
  addTodo,
  toggleTodo,

  setToggleFetching,

  setFetchedTodos,
  setAddedTodo,
  setToggledTodo,
} from '../../../client/redux/actions';

describe('actions', () => {
  const filter = 'filterName';
  const error = new Error('Boom!');
  const response = 'responseData';

  it('fetchTodos', () => {
    expect(fetchTodos()).toMatchSnapshot();
  });

  it('addTodo', () => {
    expect(addTodo('hello!')).toMatchSnapshot();
  });

  it('toggleTodo', () => {
    expect(toggleTodo('123')).toMatchSnapshot();
  });

  it('setToggleFetching', () => {
    expect(setToggleFetching(filter)).toMatchSnapshot();
    expect(setToggleFetching(error)).toMatchSnapshot();
  });

  it('setFetchedTodos', () => {
    expect(setFetchedTodos(response, filter)).toMatchSnapshot();
    expect(setFetchedTodos(error, filter)).toMatchSnapshot();
  });

  it('setAddedTodo', () => {
    expect(setAddedTodo(response)).toMatchSnapshot();
    expect(setAddedTodo(error)).toMatchSnapshot();
  });

  it('setToggledTodo', () => {
    expect(setToggledTodo(response)).toMatchSnapshot();
    expect(setToggledTodo(error)).toMatchSnapshot();
  });
});

