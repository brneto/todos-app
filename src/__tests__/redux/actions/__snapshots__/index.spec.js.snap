// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addTodo 1`] = `
Object {
  "payload": "hello!",
  "type": "ADD_TODO",
}
`;

exports[`actions fetchTodos 1`] = `
Object {
  "type": "FETCH_TODOS",
}
`;

exports[`actions setAddedTodo 1`] = `
Object {
  "payload": "responseData",
  "type": "SET_ADDED_TODO",
}
`;

exports[`actions setAddedTodo 2`] = `
Object {
  "error": true,
  "payload": [Error: Boom!],
  "type": "SET_ADDED_TODO",
}
`;

exports[`actions setFetchedTodos 1`] = `
Object {
  "meta": Object {
    "filter": "filterName",
  },
  "payload": "responseData",
  "type": "SET_FETCHED_TODOS",
}
`;

exports[`actions setFetchedTodos 2`] = `
Object {
  "error": true,
  "meta": Object {
    "filter": "filterName",
  },
  "payload": [Error: Boom!],
  "type": "SET_FETCHED_TODOS",
}
`;

exports[`actions setToggleFetching 1`] = `
Object {
  "payload": "filterName",
  "type": "SET_TOGGLE_FETCHING",
}
`;

exports[`actions setToggleFetching 2`] = `
Object {
  "error": true,
  "payload": [Error: Boom!],
  "type": "SET_TOGGLE_FETCHING",
}
`;

exports[`actions setToggledTodo 1`] = `
Object {
  "payload": "responseData",
  "type": "SET_TOGGLED_TODO",
}
`;

exports[`actions setToggledTodo 2`] = `
Object {
  "error": true,
  "payload": [Error: Boom!],
  "type": "SET_TOGGLED_TODO",
}
`;

exports[`actions toggleTodo 1`] = `
Object {
  "payload": "123",
  "type": "TOGGLE_TODO",
}
`;
